ZuluContent/Misc/ClientVerification.cs

```diff
@@ -7,172 +7,171 @@
+            }
+
+            if (kickMessage != null)
+            {
+                state.Mobile.SendMessage(0x22, kickMessage);
+                state.Mobile.SendMessage(0x22, "You will be disconnected in {0} seconds.", KickDelay.TotalSeconds);
+
+                Timer.DelayCall(KickDelay, () =>
+                {
+                    if (state.Connection != null)
+                    {
+                        state.Disconnect("Client: {0}: Disconnecting, bad version");
+                    }
+                });
+            }
+            else if (Required != null && version < Required)
+            {
+                switch (OldClientResponse)
+                {
+                    case OldClientResponseType.Warn:
+                    {
+                        state.Mobile.SendMessage(0x22, "Your client is out of date. Please update your client.",
+                            Required);
+                        state.Mobile.SendMessage(0x22,
+                            "This server recommends that your client version be at least {0}.",
+                            Required);
+                        break;
+                    }
+                    case OldClientResponseType.LenientKick:
+                    case OldClientResponseType.Annoy:
+                    {
+                        SendAnnoyGump(state.Mobile);
+                        break;
+                    }
+                }
+            }
         }
-        else if (AllowGod && !AllowRegular && !AllowUOTD && version.Type != ClientType.God)
+
+        private static void KickMessage(Mobile from, bool okay)
         {
-          kickMessage = "This server requires you to use the god client.";
+            from.SendMessage("You will be reminded of this again.");
+
+            if (OldClientResponse == OldClientResponseType.LenientKick)
+                from.SendMessage(
+                    "Old clients will be kicked after {0} days of character age and {1} hours of play time",
+                    AgeLeniency, GameTimeLeniency);
+
+            Timer.DelayCall(TimeSpan.FromMinutes(Utility.Random(5, 15)), () => SendAnnoyGump(from));
         }
-        else if (kickMessage != null)
+
+        private static void SendAnnoyGump(Mobile m)
         {
-          if (AllowRegular && AllowUOTD)
-            kickMessage += " You can use regular or UO:TD clients.";
-          else if (AllowRegular)
-            kickMessage += " You can use regular clients.";
-          else if (AllowUOTD)
-            kickMessage += " You can use UO:TD clients.";
+            if (m.NetState != null && m.NetState.Version < Required)
+            {
+                Gump g = new WarningGump(1060637, 30720,
+                    $"Your client is out of date. Please update your client.<br>This server recommends that your client version be at least {Required}.<br> <br>You are currently using version {m.NetState.Version}.<br> <br>To patch, run UOPatch.exe inside your Ultima Online folder.",
```

Comment:
we should probably remember to change this message at some point to use the launcher and not UOPatch 😛